qlty-check/src/executor/invocation_result.rs

```diff
@@ -280,9 +280,18 @@ impl InvocationResult {
     fn handle_output_rewrite(&mut self) -> Result<()> {
         let mut formatted = vec![];
 
-        for workspace_entry in self.plan.workspace_entries.iter() {
-            let workspace_path = self.plan.workspace.root.join(&workspace_entry.path);
-            let staged_path = self.plan.target_root.join(&workspace_entry.path);
+        for workspace_entry in self.invocation.target_paths.iter() {
+            let prefixed_path = if let Some(prefix) = &self.plan.plugin.prefix {
+                PathBuf::from(prefix).join(workspace_entry)
+            } else {
+                PathBuf::from(&workspace_entry)
+            };
```

Comment:
There's a pattern we seem to do twice in this PR which is to prefix-or-not a path based on a plugin definition's prefix field.

I would extract an and reuse function which accepts a plugin definition and a path and performs this operation. (I would _not_ put this function on PluginDef itself because that is intended to be a data transfer object or a configuration object and not get a bunch of behavior added to the impl)

Reply:
While doing this, realized some of the variable naming was confusing and there was another bug hiding in there that effected file names when reporting a formatters issues during check run.